ttnn/cpp/ttnn/operations/ccl/reduce_scatter/reduce_scatter_op.hpp

```diff
@@ -12,14 +12,14 @@ namespace operations {
 namespace ccl {
 
 struct ExecuteReduceScatter {
```

Comment:
this file should be named 
ttnn/cpp/ttnn/operations/ccl/reduce_scatter/reduce_scatter.hpp

Reply:
and I'd prefer `reduce_scatter` method to be gone from 
ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/reduce_scatter_op.hpp
and its implementation to be in reduce_scatter.cpp

also not sure why we need reduce_scatter_impl

Reply:
renamed file to reduce_scatter.hpp but haven't applied this other change. When I apply it as I understand it, there are some issues with device/tensor identification  that lead to a failure. I'll investigate it tomorrow.

Reply:
I dug into the change a little where we remove the explicit launch op and instead register the op with `register_operation_with_auto_launch_op`. Unfortunately this leads to further issues which haven't been root caused yet. 

Some of this seems to be due to missing codepaths for support for `MultiDeviceStorage` storage types in some of the ttnn operations infra. I added these paths where I suspected they were missing (obviously pending review) to unblock and make further progress. 

However, I hit another issue that seems to be taking me down a rabbit hole. At this point I am going to open an issue  to enable this functionality for CCL ops with mesh tensors. I don't think this has been done yet and so I believe I'm path clearing some unrelated (to reduce scatter specifically) issues in the general op infra.

The next issue I am seeing is that querying the mesh tensor for workers returns the correct number of workers for the (I assume) first chip but for the second (and beyond) chips where we try to generate a reduce scatter program, we are only getting a single worker, so we aren't able to identify our neighbour device IDs (in this case in the multi-device ring).

So I'm going to push back and say that I will for now stay with using the explicit `LaunchOp` approach and perhaps let's revisit this when some cycles open up.

Reply:
Issue tracked here: https://github.com/tenstorrent/tt-metal/issues/10756

Reply:
merged reduce_scatter_impl + reduce_scatter.